src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import classNames from 'classnames';
+import {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import { Todo } from '../../types/Todo';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  todos: Todo[],
+  setErrorMessage: (error: Errors) => void,
+  updateTodo: (updatedTodo: Todo) => Promise<void>,
+  addTodo: (inputValue: string) => Promise<void>,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  todos,
+  setErrorMessage,
+  updateTodo,
+  addTodo,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const headerInputFocus = useRef<HTMLInputElement | null>(null);
+
+  // Autofocus on main input
+  useEffect(() => {
```

Comment:
The useEffect hook has 'todos' as a dependency and will re-run every time 'todos' changes. This could lead to unnecessary re-renders and affect performance. Consider alternative approaches, like using a callback ref to handle focusing.